### PR TITLE
IFTAQCAA-27 - test script to cover vacancies page functionality

### PR DESCRIPTION
**Pull request by dkrutenko**
verify vacancy page functionality

### CODE REVIEW

File: src/main/java/com/softserveinc/ita/pageobjects/modals/UserModal.java [hunk 1]
```diff
@@ -1,7 +1,18 @@
 package com.softserveinc.ita.pageobjects.modals;
 
-import com.softserveinc.ita.pageobjects.BasePage;
+import lombok.Builder;
+import lombok.Getter;
+
+@Builder
+@Getter
+public class UserModal {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/modals/UserModal.java:8]: so, if it's model, rename to User and move into models package

File: src/main/java/com/softserveinc/ita/pageobjects/modals/UserModal.java [hunk 2]
```diff
@@ -1,7 +1,18 @@
...
+
+    private String firstName;
+    private String lastName;
+    private String phone;
+    private String email;
+    private String categoryValue;
+    private String departamentValue;
 
-public class UserModal extends BasePage<UserModal> {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/modals/UserModal.java:17]: delete empty lines

File: src/main/java/com/softserveinc/ita/pageobjects/HomePage.java [hunk 3]
```diff
@@ -76,4 +76,11 @@ public ProductPage openLastViewedItemByTitle(String expectedItem) {
 
         return new ProductPage();
     }
+
+    @Step("HomePage: Opened sidebar page '{expectedPage}' by name")
+    public HomePage openSidebarPage(String expectedPage) {
+        $x("//li[@class='main-links__item ng-star-inserted']/a[contains(text(),'"+expectedPage+"')]").click();
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/HomePage.java:82]: use String.format()

File: src/main/java/com/softserveinc/ita/utils/ElementsUtil.java [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package com.softserveinc.ita.utils;
+
+import com.codeborne.selenide.SelenideElement;
+import lombok.experimental.UtilityClass;
+import org.openqa.selenium.support.ui.Select;
+
+@UtilityClass
+public class ElementsUtil {
+
+    public SelenideElement setText(SelenideElement elementSelector, String text) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/ElementsUtil.java:10]: void

File: src/main/java/com/softserveinc/ita/utils/ElementsUtil.java [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+        var input = elementSelector;
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/ElementsUtil.java:11]: you don't need it, you could use element from method argument

File: src/main/java/com/softserveinc/ita/utils/ElementsUtil.java (hunk 4 above)
Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/utils/ElementsUtil.java:10]: elementSelector -> webElement

File: src/main/java/com/softserveinc/ita/utils/ElementsUtil.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+        input.click();
+        input.sendKeys(text);
+
+        return input;
+    }
+
+    public void selectModalMenu(SelenideElement selector , String text) {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/ElementsUtil.java:18]: selector -> webElement

Comment (Nikodych) [reply src/main/java/com/softserveinc/ita/utils/ElementsUtil.java:18]: selectOptionFromMenu

File: src/main/java/com/softserveinc/ita/utils/ElementsUtil.java [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+        var selectForm = selector;
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/utils/ElementsUtil.java:19]: same, you have this element in arguments

File: src/main/java/com/softserveinc/ita/pageobjects/BasePage.java [hunk 8]
```diff
@@ -55,10 +55,10 @@ public SearchResultPage performSearch() {
     }
 
     @Step("BasePage: Opened user modal window")
-    public UserModal openUserModalWindow() {
+    public T openUserModalWindow() {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/BasePage.java:58]: here I'm not sure, is it used somewhere? if we know that it should open user modal, so probably here should returning page object which is referring to the user

File: src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+        return this;
+    }
+
+    public VacancyPage clickOnSendDataButton() {
+        $x("//button[@class='button button--green button--medium want-work__submit ng-star-inserted']").click();
+
+        return this;
+    }
+
+    public VacancyPage fillVacancyPageFields() {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java:32]: fillVacancyPageFields(User user)

File: src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+        setText($x("//input[@id='first_name']"), getVacancyUser().getFirstName());
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java:33]: user.getFirstName()

File: src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java [hunk 11]
```diff
@@ -0,0 +1,57 @@
...
+        setText($x("//input[@id='last_name']"), getVacancyUser().getLastName());
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java:34]: same for all

File: src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+        setText($x("//input[@id='phone']"), getVacancyUser().getPhone());
+        setText(emailFieldSelector, getVacancyUser().getEmail());
+
+        return this;
+    }
+
+    public VacancyPage selectCategoryAndDepartament() {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java:41]: selectCategoryAndDepartament(User user)

File: src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java [hunk 13]
```diff
@@ -0,0 +1,57 @@
...
+        selectModalMenu($x("//div[@class='form__row ng-star-inserted'][1]/select"), getVacancyUser().getCategoryValue());
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java:42]: same as above

File: src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java [hunk 14]
```diff
@@ -0,0 +1,57 @@
...
+
+        return this;
+    }
+
+    public String getAttributeFromEmailField() {
+        return emailFieldSelector.getAttribute("value").trim();
+    }
+
+    public String getTextFromCategoryModal(String categoryId) {
+        return $x("//*[@id='categories']/option["+categoryId+"]")
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java:53]: use String.format()

File: src/test/java/com/softserveinc/ita/HomePageTest.java [hunk 15]
```diff
@@ -107,4 +108,35 @@ public void verifyOpeningMobileAppLinksTest() {
...
+
+    @DataProvider
+    public Object[][] vacancyPageData() {
+        return new Object[][]{
+                {"Вакансії", "dospecwork@gmail.com", "25", "QA engineers"}};
+    }
+
+    @Test(dataProvider = "vacancyPageData")
+    public void verifyVacancyPageFunctionality(String expectedPage, String expectedEmail, String categoryId, String expectedCategory) {
+        homePage
```
Comment (Nikodych) [inline src/test/java/com/softserveinc/ita/HomePageTest.java:121]: create here var user = getVacancyUser();

and pass it into methods

**Review by Nikodych — CHANGES_REQUESTED** [review]

File: src/main/java/com/softserveinc/ita/utils/ElementsUtil.java (hunk 7 above)
Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/utils/ElementsUtil.java:19]: its deprecated class

File: src/main/java/com/softserveinc/ita/pageobjects/BasePage.java (hunk 8 above)
Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/pageobjects/BasePage.java]: removed non-usable method

File: src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java (hunk 14 above)
Comment (dkrutenko) [reply src/main/java/com/softserveinc/ita/pageobjects/VacancyPage.java]: fixed

File: src/main/java/com/softserveinc/ita/pageobjects/modals/User.java [hunk 16]
```diff
@@ -5,14 +5,12 @@
 
 @Builder
 @Getter
-public class UserModal {
+public class User {
```
Comment (Nikodych) [inline src/main/java/com/softserveinc/ita/pageobjects/modals/User.java:8]: move to models package

**Review by Nikodych — CHANGES_REQUESTED** [review]

**Comment by Nikodych** [comment]
class User is still not in that package

**Comment by dkrutenko** [comment]
How ?
image

**Review by Nikodych — APPROVED** [review]

**Pull request merged**